core/storage/email/gae_models.py

```diff
@@ -438,3 +438,169 @@ def thread_id(self):
+        Args:
+            reply_to_id: str. The unique 'reply-to' id.
+
+        Returns:
+            str or None. The FeedbackEmailReplyToIdModel instance corresponding
+                to the given 'reply-to' id if it is fetched else None.
+        """
+        model = cls.query(cls.reply_to_id == reply_to_id).fetch()
+        if not model:
+            return None
+        return model[0]
+
+    @classmethod
+    def get(cls, user_id, thread_id, strict=True):
+        """Gets the FeedbackEmailReplyToIdModel instance corresponding to the
+        unique instance id.
+
+        Args:
+            user_id: str. The user id.
+            thread_id: str. The thread id.
+            strict: bool. Whether to fail noisily if no entry with the given
+                instance id exists in the datastore. Default is True.
+
+        Returns:
+            A FeedbackEmailReplyToIdModel instance that corresponds to the given
+                instance id if it is present in the datastore. Otherwise, None.
+        """
+        instance_id = cls._generate_id(user_id, thread_id)
+        return super(
+            GeneralFeedbackEmailReplyToIdModel, cls).get(
+                instance_id, strict=strict)
+
+    @classmethod
+    def get_multi_by_user_ids(cls, user_ids, thread_id):
+        """Returns the FeedbackEmailReplyToIdModel instances corresponding to
+        the given user ids in dict format.
+
+        Args:
+            user_ids: list(str). A list of user ids.
+            thread_id: str. The thread id.
+
+        Returns:
+            dict. The FeedbackEmailReplyToIdModel instances corresponding to the
+                given list of user ids in dict format. The key is the unique
+                user id, and the corresponding value is the list of
+                FeedbackEmailReplyToIdModel instances.
+        """
+        instance_ids = [cls._generate_id(user_id, thread_id)
+                        for user_id in user_ids]
+        user_models = cls.get_multi(instance_ids)
+        return {
+            user_id: model for user_id, model in zip(user_ids, user_models)}
+
+    @property
+    def user_id(self):
+        """Returns the user id corresponding to this FeedbackEmailReplyToIdModel
+        instance.
+
+        Returns:
+            str. The user id.
+        """
+        return self.id.split('.')[0]
+
+
+    @property
+    def entity_type(self):
+        """Returns the entity type extracted from the unique id.
+
+        Returns:
+            str. The entity type.
+        """
+        return self.id.split('.')[1]
+
+
+    @property
+    def entity_id(self):
+        """Returns the entity id extracted from the unique id.
+
+        Returns:
+            str. The entity id.
+        """
+        return self.id.split('.')[2]
+
+    @property
+    def thread_id(self):
+        """Returns the thread id extracted from the unique id.
+
+        Returns:
+            str. The thread id.
+        """
+        return self.id[self.id.find('.') + 1:]
```

Comment:
This seems a bit icky. Better to split and then build it up, or do something like '.'.join(self.entity_type, self.entity_id) (if I've got it right). It makes it clearer how the thing is constructed.

Reply:
Done